src/components/Context/createComponent.js

```diff
@@ -28,8 +28,9 @@ export default (config, logger, availableContexts, requiredContexts) => {
     lifecycle: {
       onBeforeEvent({ event }) {
         const xdm = {};
-        contexts.forEach(context => context(xdm));
-        event.mergeXdm(xdm);
+        return Promise.all(
+          contexts.map(context => Promise.resolve(context(xdm)))
+        ).then(event.mergeXdm(xdm));
```

Comment:
Now that I'm thinking about this, `.then(event.mergeXdm(xdm))` should probably be `.then(() => event.mergeXdm(xdm))`, unless you want `event.mergeXdm(xdm)` to run immediately and it returns a function that should be run once the previous line resolves.

Reply:
I think this should be a little bit refactored.